pkg/playground/dick.tsx

```diff
@@ -0,0 +1,557 @@
+                    label={checkbox_label}
+                    onChange={() => value.set(false)}
+                />
+                <DialogTextInput value={value.at(val)} />
+            </>
+        );
+    }
+
+    return (
+        <FormGroup
+            label={field_label}
+        >
+            {body}
+            <DialogHelperText value={value} />
+        </FormGroup>
+    );
+};
+
+function async_sleep(n: number) {
+    return new Promise(resolve => {
+        window.setTimeout(resolve, n);
+    });
+}
+
+interface Color {
+    name: string,
+    red: number,
+    green: number,
+    blue: number,
+}
+
+const colors: Color[] = [
+    { name: "red", red: 1, green: 0, blue: 0 },
+    { name: "green", red: 0, green: 1, blue: 0 },
+    { name: "blue", red: 0, green: 0, blue: 1 },
+];
+
+interface ExampleValues {
+    flag: boolean;
+    text: string;
+    radio: string;
+    dropdown: string;
+    color: Color,
+    list: string[];
+    async: Name[];
+    alternative: false | string;
+    error: string;
+}
+
+const ExampleDialog = ({
+    setResult,
+    countAsyncValidation,
+} : {
+    setResult: (values: ExampleValues) => void,
+    countAsyncValidation: () => void,
+}) => {
+    const Dialogs = useDialogs();
+
+    function init(): ExampleValues {
+        return {
+            flag: false,
+            text: "",
+            radio: "one",
+            dropdown: "one",
+            color: colors[0],
+            list: [],
+            async: [],
+            alternative: false,
+            error: "none",
+        };
+    }
+
+    function validate(dlg: DialogState<ExampleValues>) {
+        if (dlg.values.flag) {
+            dlg.value("text").validate(v => {
+                if (!v)
+                    return "Text can not be empty";
+            });
+        }
+        dlg.value("list").forEach(v => {
+            v.validate(vv => {
+                if (vv == ".")
+                    return "No dots";
+            });
+        });
+        dlg.value("async").forEach(v => validate_Name(v, countAsyncValidation));
+    }
+
+    const dlg = useDialogState(init, validate);
+
+    async function apply(values: ExampleValues) {
+        setResult(values);
+
+        if (values.error == "custom") {
+            throw new DialogError("This is a failure", <code>1234-567-98A</code>);
+        } else if (values.error == "from") {
+            const err = new Error("no such file or scraper");
+            throw DialogError.fromError("Tool not found", err);
+        } else if (values.error == "from-random") {
+            const err = [1, 2, 3, 4];
+            throw DialogError.fromError("Too random", err);
+        } else if (values.error == "message") {
+            // eslint-disable-next-line no-throw-literal
+            throw { message: "segmentation fault" };
+        } else if (values.error == "spawn") {
+            await cockpit.spawn(["ls", "--no-such-option"], { err: "message" });
+        } else if (values.error == "random") {
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22700-6940558b-20251219-144917-fedora-43-devel-other/Coverage/lcov/github-pr.diff.gcov.html)